staging/src/k8s.io/apiserver/pkg/storage/feature/feature_support_checker.go

```diff
@@ -43,67 +47,100 @@ type FeatureSupportChecker interface {
+	checkingEndpoint           map[string]struct{}
+	clock                      clock.Clock
 }
 
 func newDefaultFeatureSupportChecker() *defaultFeatureSupportChecker {
 	return &defaultFeatureSupportChecker{
 		progresNotifyEndpointCache: make(map[string]bool),
+		checkingEndpoint:           make(map[string]struct{}),
+		clock:                      clock.RealClock{},
 	}
 }
 
 // Supports can check the featue from anywhere without storage if it was cached before.
-func (f *defaultFeatureSupportChecker) Supports(feature storage.Feature) (bool, error) {
+func (f *defaultFeatureSupportChecker) Supports(feature storage.Feature) bool {
 	switch feature {
 	case storage.RequestWatchProgress:
 		f.lock.Lock()
 		defer f.lock.Unlock()
 
-		return ptr.Deref(f.progressNotifySupported, false), nil
+		return ptr.Deref(f.progressNotifySupported, false)
 	default:
-		return false, fmt.Errorf("feature %q is not implemented in DefaultFeatureSupportChecker", feature)
+		klog.Errorf("feature %q is not implemented in DefaultFeatureSupportChecker", feature)
+		return false
 	}
 }
 
 // CheckClient accepts client and calculate the support per endpoint and caches it.
-// It will return at any point if error happens or one endpoint is not supported.
-func (f *defaultFeatureSupportChecker) CheckClient(ctx context.Context, c client, feature storage.Feature) error {
+func (f *defaultFeatureSupportChecker) CheckClient(ctx context.Context, c client, feature storage.Feature) {
 	switch feature {
 	case storage.RequestWatchProgress:
-		return f.clientSupportsRequestWatchProgress(ctx, c)
+		f.checkClient(ctx, c)
 	default:
-		return fmt.Errorf("feature %q is not implemented in DefaultFeatureSupportChecker", feature)
-
+		klog.Errorf("feature %q is not implemented in DefaultFeatureSupportChecker", feature)
 	}
 }
 
-func (f *defaultFeatureSupportChecker) clientSupportsRequestWatchProgress(ctx context.Context, c client) error {
+func (f *defaultFeatureSupportChecker) checkClient(ctx context.Context, c client) {
+	// start with 10 ms, multiply by 2 each step, until 15 s and stays on 15 seconds.
+	backoff := wait.Backoff{
+		Duration: 10,
+		Cap:      15 * time.Second,
+		Factor:   2.0,
+		Jitter:   0.0,
+		Steps:    50}.DelayWithReset(f.clock, 15*time.Second)
 	f.lock.Lock()
 	defer f.lock.Unlock()
-
 	for _, ep := range c.Endpoints() {
-		supported, err := f.supportsProgressNotifyEndpointLocked(ctx, c, ep)
-		if err != nil {
-			return err
-		}
-		if !supported {
-			f.progressNotifySupported = ptr.To(false)
-			return nil
+		if _, found := f.checkingEndpoint[ep]; found {
+			continue
 		}
+		f.checkingEndpoint[ep] = struct{}{}
+		go func(ep string) {
+			err := backoff.Until(ctx, true, true, func(ctx context.Context) (done bool, err error) {
+				defer runtime.HandleCrash()
+				internalErr := f.clientSupportsRequestWatchProgress(ctx, c, ep)
+				return internalErr != nil, nil
+			})
+			if err != nil {
+				klog.ErrorS(err, "Failed to check client support after retrying")
```

Comment:
consider changing the text to `Failed to check if RequestWatchProgress is supported by etcd after retrying` or something similar that would give more context.

Reply:
In addition we could use `utilruntime.HandleError`